### PR TITLE
Extension names 531

### PR DESCRIPTION
**Pull request by 007vasy**
Test for catching [this bug](https://github.com/FHIR/sushi/issues/531)

### CODE REVIEW

**Comment by 007vasy** [comment]
@ngfreiter can you have a look?

File: test/export/StructureDefinitionExporter.test.ts [hunk 1]
```diff
@@ -343,6 +343,32 @@ describe('StructureDefinitionExporter', () => {
...
+    const FooFooCardRule = new CardRule('extension[FooFoo]');
+    FooFooCardRule.min = 1;
+    FooFooCardRule.max = '1'; // * extension[sliceB].extension 1..1
+
+    const FooBarCardRule = new CardRule('extension[FooBar]');
+    FooBarCardRule.min = 0;
+    FooBarCardRule.max = '1'; // * extension[sliceB].extension 0..0
+
+    const containsRule = new ContainsRule('extension');
+    containsRule.items = [{ name: 'FooFoo' }, { name: 'FooBar' }];
```
Comment (ngfreiter) [inline test/export/StructureDefinitionExporter.test.ts:362]: Thanks for the test! I think I know why this test isn't catching the original bug. In the original report, the first extension name, `forward` is entirely contained within the second extension name, `forwardByteSize`. In this test, that is not the case, `FooFoo`  is not a prefix of `FooBar`. If you change this test so that the `FooFoo` extension is just named `Foo`, so it has a name that is a prefix of `FooBar`, then the test should correctly fail.

File: 531.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+ npm run test test/export/StructureDefinitionExporter.tests
```
Comment (ngfreiter) [inline 531.md:1]: For this to be ready to merge, we will want to remove this file.

File: README.md [hunk 3]
```diff
@@ -123,4 +123,4 @@ Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
-limitations under the License.
```
Comment (ngfreiter) [inline README.md:126]: We'd want to undo this change as well, looks like a whitespace change.

File: test/export/StructureDefinitionExporter.test.ts [hunk 4]
```diff
@@ -343,6 +343,32 @@ describe('StructureDefinitionExporter', () => {
     expect(exported.derivation).toBe('constraint');
   });
 
+  it('should export sub-extensions, with similar starting names and different types', () => {
```
Comment (ngfreiter) [inline test/export/StructureDefinitionExporter.test.ts:346]: Once this test is correctly failing, can you skip it using `it.skip`? That way we can merge in the failing test to master before pushing up the fix. Then, when we have a bug fix ready, we can un-skip the test.

**Comment by 007vasy** [comment]
@cmoesel
